presto/dbapi.py

```diff
@@ -281,17 +282,44 @@ def _prepare_statement(self, operation, statement_name):
 
         raise presto.exceptions.FailedToObtainAddedPrepareHeader
 
+
     def _get_added_prepare_statement_presto_query(
         self,
         statement_name,
         params
     ):
-        sql = 'EXECUTE ' + statement_name + ' USING ' + ','.join(map(str, params))
+        sql = 'EXECUTE ' + statement_name + ' USING ' + ','.join(map(self._format_prepared_param, params))
 
         # No need to deepcopy _request here because this is the actual request
         # operation
         return presto.client.PrestoQuery(self._request, sql=sql)
 
+    def _format_prepared_param(self, param):
+        """
+        Formats parameters to be passed in an
+        EXECUTE statement.
+        """
+        if isinstance(param, bool):
+            return "true" if param else "false"
+        if isinstance(param, float):
+            return "DOUBLE '%s'" % param
+        if isinstance(param, str):
+            return ("'%s'" % param.replace("'", "''"))
+        if isinstance(param, bytes):
+            return "X'%s'" % param.hex()
+        if isinstance(param, datetime.datetime):
+            datetime_str = param.strftime("%Y-%m-%d %H:%M:%S.%f %Z").rstrip(" ")
+            return "TIMESTAMP '%s'" % re.sub(r'\.([0-9]{3})([0-9]{3})', r'.\1', datetime_str)
```

Comment:
prior to presto 335 the timestamp format only supports 1/100th of a second in the timestamp format, thus the last 3 digits of the subseconds are truncated (effectively dividing by 1000).

Reply:
we can have a legacy mode for dealing with older versions, but we should be correct by default (no implicit truncation nor rounding) for the new versions

doing truncation via regular expression is very clever, but i am thinking we could perhaps round using some methods on `param`, or by providing desired output format to `strftime` directly.

Reply:
just to clarify, should we only support the later versions here and forego truncating the subseconds? or round depending on how the user constructed the `datetime`?

Reply:
I do not see a need to support old versions here.

we can always revisit if there is enough demand

Reply:
removed truncation of subseconds